src/rust/ensogl/lib/components/src/shadow.rs

```diff
@@ -90,3 +97,31 @@ pub fn add_to_dom_element(element:&DomSymbol, style:&StyleWatch,logger:&Logger)
     let shadow = format!("{}px {}px {}px {}px rgba(0,0,0,{})",off_x,off_y,blur,spread,alpha);
     element.dom().set_style_or_warn("box-shadow",shadow,&logger);
 }
+
+
+/// Provides FRP endpoints for the parameters that define the appearance of a shadow
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct ParametersFrp {
+    pub base_color : frp::Sampler<color::Rgba>,
+    pub fading     : frp::Sampler<color::Rgba>,
+    pub size       : frp::Sampler<f32>,
+    pub spread     : frp::Sampler<f32>,
+    pub exponent   : frp::Sampler<f32>,
+    pub offset_x   : frp::Sampler<f32>,
+    pub offset_y   : frp::Sampler<f32>
+}
+
+/// Return FRP endpoints for the parameters that define a shadow.
+pub fn frp_from_style(style:&StyleWatchFrp,path:impl Into<style::Path>) -> ParametersFrp {
+    let path: style::Path = path.into();
+    ParametersFrp{
+        base_color: style.get_color(&path),
+        fading: style.get_color(&path.sub("fading")),
+        size: style.get_number(&path.sub("size")),
+        spread: style.get_number(&path.sub("spread")),
+        exponent: style.get_number(&path.sub("exponent")),
+        offset_x: style.get_number(&path.sub("offset_x")),
+        offset_y: style.get_number(&path.sub("offset_y"))
+    }
+}
```

Comment:
I don't see any usage of this structure, and it's likely to be not tested. Do you insist to merging this anyway? Or you wanted to use it?

Reply:
It is used in `ensogl/lib/components/src/selector/common/base_frp.rs:61:33` and it will be used more heavily in the style refactoring that will replace most usages of the non-frp version `shadow::Parameters` when implementing #795